fluss-server/src/main/java/com/alibaba/fluss/server/kv/mergeengine/VersionRowMergeEngine.java

```diff
@@ -0,0 +1,119 @@
+
+    public static final Set<String> VERSION_MERGE_ENGINE_SUPPORTED_DATA_TYPES =
+            ImmutableSet.of(
+                    BigIntType.class.getName(),
+                    IntType.class.getName(),
+                    TimestampType.class.getName(),
+                    TimeType.class.getName(),
+                    LocalZonedTimestampType.class.getName());
+
+    private final int fieldIndex;
+    private final InternalRow.FieldGetter fieldGetter;
+    private final RowType rowType;
+
+    public VersionRowMergeEngine(Schema schema, MergeEngine mergeEngine) {
+        this.rowType = schema.toRowType();
+        InternalRow.FieldGetter[] currentFieldGetters =
+                new InternalRow.FieldGetter[rowType.getFieldCount()];
+        for (int i = 0; i < rowType.getFieldCount(); i++) {
+            currentFieldGetters[i] = InternalRow.createFieldGetter(rowType.getTypeAt(i), i);
+        }
+        this.fieldIndex = rowType.getFieldIndex(mergeEngine.getColumn());
+        if (fieldIndex == -1) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "The merge engine is set to version, but the version column %s does not exist.",
+                            mergeEngine.getColumn()));
+        }
+        fieldGetter = currentFieldGetters[fieldIndex];
+    }
+
+    @Override
+    public BinaryRow merge(BinaryRow oldRow, BinaryRow newRow) {
+        Object oldValue = fieldGetter.getFieldOrNull(oldRow);
+        Object newValue = fieldGetter.getFieldOrNull(newRow);
+        // If the new value is empty, ignore it directly.
+        if (newValue == null) {
+            return null;
+        }
+        // If the old value is null, simply overwrite it with the new value.
+        if (oldValue == null) {
+            return newRow;
+        }
+        DataType dataType = rowType.getTypeAt(fieldIndex);
+        return getValueComparator(dataType).isGreaterThan(newValue, oldValue) ? newRow : null;
+    }
+
+    @Override
+    public boolean ignoreDelete() {
+        return true;
+    }
+
+    @Override
+    public boolean ignorePartialUpdate() {
+        return true;
+    }
+
+    private ValueComparator getValueComparator(DataType dataType) {
+        if (dataType instanceof BigIntType) {
+            return (left, right) -> (Long) left > (Long) right;
+        }
+        if (dataType instanceof IntType || dataType instanceof TimeType) {
+            return (left, right) -> (Integer) left > (Integer) right;
+        }
+        if (dataType instanceof TimestampType) {
+            return (left, right) ->
+                    ((TimestampNtz) left).getMillisecond()
+                            > ((TimestampNtz) right).getMillisecond();
+        }
+        if (dataType instanceof LocalZonedTimestampType) {
+            return (left, right) ->
+                    ((TimestampLtz) left).toEpochMicros() > ((TimestampLtz) right).toEpochMicros();
```

Comment:
Should use `TimestampLtz#compareTo` to consider nano seconds. Add tests for `TIMESTAMP_LTZ(9)`.